game/missiongenerator/aircraft/flightgroupspawner.py

```diff
@@ -142,6 +142,11 @@ def generate_mid_mission(self) -> FlyingGroup[Any]:
         if alt_type == "BARO" and alt < MINIMUM_MID_MISSION_SPAWN_ALTITUDE:
             alt = MINIMUM_MID_MISSION_SPAWN_ALTITUDE
 
+        # If 'alt' is less than 500ft AGL, use random value between 500-1000ft
+        # Otherwise planes might crash in trees and stuff...
+        if alt_type == "RADIO" and alt < feet(500):
```

Comment:
Extract a constant the way the baro altitude does (rename the baro one to end `_MSL` to differentiate).

Drop the altitude randomization. If that's a real problem, let's handle it separately because in not convinced this is the correct fix.

Where are the AGL altitudes even coming from? Is that something unique to helicopters? I thought our flight plans were fully MSL (there was a pretty severe bug where the AI could not fly AGL without getting stuck at high AoA).

Reply:
One more thing though, if randomizing the AGL altitude is no good, then why is it done for position??? 🤔